xen/arch/arm/tee/ffa_partinfo.c

```diff
@@ -197,8 +217,10 @@ static void uninit_subscribers(void)
 {
         subscr_vm_created_count = 0;
         subscr_vm_destroyed_count = 0;
-        XFREE(subscr_vm_created);
-        XFREE(subscr_vm_destroyed);
+        if ( subscr_vm_created )
```

Comment:
I'm not sure these tests are needed, but if they are please put the changes in a separate commit.

Reply:
uninit subscribers is now called even if there is no firmware or the firmware does support part_info_get so we can come in a situation where the pointer is NULL and was not allocated.
This was not possible before this change.

Reply:
They are still initialized to NULL. Why does it matter if it was allocated before or not? Is it a style issue?

Reply:
right my bad, xfree is in fact just doing a return in case of NULL. Will remove